climate_learn/models/components/resnet.py

```diff
@@ -82,15 +82,32 @@ def __init__(
         if prob_type == 'parametric':
             self.final_std = PeriodicConv2D(hidden_channels, out_channels, kernel_size=7, padding=3)
 
-    def predict(self, x):
+        # final layer for categorical, change to be number of bins
+        if self.prob_type == 'categorical':            
+            self.final = PeriodicConv2D(self.hidden_channels, 50, kernel_size=7, padding=3)
```

Comment:
The second argument here should be parameterized that can be passed in at initialization (e.g., `self.num_output_bins`).

Reply:
resolved.